src/checkers/inference/InferenceLauncher.java

```diff
@@ -0,0 +1,355 @@
+package checkers.inference;
+
+
+import checkers.inference.InferenceOptions.InitStatus;
+import org.checkerframework.framework.test.TestUtilities;
+import org.checkerframework.framework.util.CheckerMain;
+import org.checkerframework.framework.util.ExecUtil;
+import org.checkerframework.framework.util.PluginUtil;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+public class InferenceLauncher {
```

Comment:
InferenceLauncher will get a comment.
You can think of InferenceLauncher as the launch script for the inference framework.  This class replaces the python script that was previously used to parse the options to the inference framework and execute the various activities needed to run inference.
InferenceLauncher can be run in 4 modes:
1) TYPECHECK: Run an inference checker in typecheck mode without inferring any annotations.
2) INFER: Run an inference checker in inference mode, which builds a set of constraints, runs a solver (which solves or serializes the previously built constraints), and outputs a JAIF file which can be used to re-insert the solved constraints.
3) ROUNDTRIP: Run inference (from mode 2) and then insert solved constraints.  While isnertion can be done using a previously output JAIF file and the Annotation File Utilities, the InferenceLauncher will allow you to insert the annotations immediately after inference.
4) ROUNDTRIP_TYPECHECK: This first runs all of the steps in ROUNDTRIP mode then runs typechecking over the newly updated code.